src/ImageSharp/Formats/Png/Filters/AverageFilter.cs

```diff
@@ -22,18 +22,26 @@ internal static unsafe class AverageFilter
         /// <param name="bytesPerScanline">The number of bytes per scanline</param>
         /// <param name="bytesPerPixel">The bytes per pixel.</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static void Decode(byte[] scanline, byte[] previousScanline, int bytesPerScanline, int bytesPerPixel)
+        public static void Decode(BufferSpan<byte> scanline, BufferSpan<byte> previousScanline, int bytesPerScanline, int bytesPerPixel)
         {
+            ref byte scanPointer = ref scanline.DangerousGetPinnableReference();
```

Comment:
Hi there! :)

Just curious how you know for sure these buffers will be pinned? If the default `ArrayPool` allocates it's slabs on LOH then it'd be safe so long as callers haven't tweaked the GC config. Not sure to be honest, I haven't actually checked that, but AFAIK the `ArrayPool` implementation in CoreFX doesn't guarantee the buffers it gives you will be pinned.

In CoreFXLab for instance, there's [a pattern for retaining a buffer](https://github.com/dotnet/corefxlab/blob/master/src/System.Buffers.Primitives/System/Internal/ManagedBufferPool.cs#L62-L71), which uses a `GcHandle` to pin if it's not expected to be already.